frame/curve-amm/src/tests.rs

```diff
@@ -229,3 +262,149 @@ fn exchange_test() {
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &ALICE, alice_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), alice_balance.into());
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &ALICE), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDC, &ALICE, alice_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &ALICE), alice_balance.into());
+
+		// Add funds to BOB's account.
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &BOB, bob_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), bob_balance.into());
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDC, &BOB, bob_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), bob_balance.into());
+
+		// Add funds to CHARLIE's account.
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &CHARLIE), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &CHARLIE, alice_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &CHARLIE), alice_balance.into());
+
+
+		// create DEX pool for 1 USDC = 1 USDT
+		let p = CurveAmm::create_pool(&ALICE, assets, amp_coeff, fee, admin_fee);
+		assert_ok!(&p);
+		let pool_id = p.unwrap();
+		let pool = CurveAmm::pool(pool_id);
+		let pool_lp_asset = CurveAmm::pool_lp_asset(pool_id);
+		assert!(pool.is_some());
+		assert!(pool_lp_asset.is_some());
+		let pool_lp_asset = pool_lp_asset.unwrap();
+
+		// ALICE adds liquidity to DEX pool.
+		let alice_amounts = vec![alice_balance as u128, alice_balance as u128];
+		assert_ok!(CurveAmm::add_liquidity(&ALICE, pool_id, alice_amounts.clone(), 0u128));
+		let alice_lp_balance = Tokens::balance(pool_lp_asset, &ALICE);
+		assert_ne!(alice_lp_balance, 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &ALICE), 0);
+
+		// BOB adds liquidity to DEX pool.
+		let bob_amounts = vec![bob_balance as u128, bob_balance as u128];
+		assert_ok!(CurveAmm::add_liquidity(&BOB, pool_id, bob_amounts.clone(), 0u128));
+		let bob_lp_balance = Tokens::balance(pool_lp_asset, &BOB);
+		assert_ne!(bob_lp_balance, 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), 0);
+
+		// CHARLIE exchanges USDT for USDC, CHARLIE has same balance of USDT as of ALICE.
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &CHARLIE), 0);
+		assert_ok!(CurveAmm::exchange(&CHARLIE, pool_id, 1, 0, alice_balance as u128, 0));
+		prop_assert_epsilon!(alice_balance as u128, Tokens::balance(MockCurrencyId::USDC, &CHARLIE));
+		Ok(())
+	}).unwrap();
+	}
+
+	#[test]
+	fn proptest_add_remove_liquidity(alice_balance in 0..u32::MAX,
+									 bob_balance in 0..u32::MAX) {
+	new_test_ext().execute_with(|| {
```

Comment:
weirdly formatted?

Reply:
On my system vim and vscode shows this line properly aligned.